### ISSUE TITLE
bytesAhead wrong when buffer is smaller that a previous one

### ISSUE THREAD

**Issue by bkw** [felixge/node-buffy#1]
bytesAhead() depends on buffer.length which is not decreased when we reuse a larger buffer.
I'm afraid we'll have to track the bytes ready to read ourselves. I'll give it a try.
In the meantime, here is a failing test: bkw/node-buffy@2363ba2b

**Comment by felixge** [comment]
Great find, yes this is broken!

> I'm afraid we'll have to track the bytes ready to read ourselves. I'll give it a try.

I think we can do something better. Instead of moving the remaining bytes to the beginning of the buffer, we should move them just far enough to the front that `newBuffer` can fit afterwards. Then instead of setting `this._offset = 0`, we set it to the new offset we calculated. I don't have time right now, but this should allow us to fix this issue without introducing additional house keeping!
